Library/Homebrew/cask/lib/hbc/dsl/depends_on.rb

```diff
@@ -84,23 +84,23 @@ def cask=(*arg)
         @cask.concat(Array(*arg))
       end
 
-      def macos=(*arg)
+      def macos=(*args)
         @macos ||= []
-        macos = if arg.count == 1 && arg.first =~ /^\s*(<|>|[=<>]=)\s*(\S+)\s*$/
+        macos = if args.count == 1 && args.first =~ /^\s*(<|>|[=<>]=)\s*(\S+)\s*$/
           raise "'depends_on macos' comparison expressions cannot be combined" unless @macos.empty?
           operator = Regexp.last_match[1].to_sym
           release = self.class.coerce_os_release(Regexp.last_match[2])
           [[operator, release]]
         else
           raise "'depends_on macos' comparison expressions cannot be combined" if @macos.first.is_a?(Symbol)
-          [*arg].map(&self.class.method(:coerce_os_release)).sort
+          Array(*args).map(&self.class.method(:coerce_os_release)).sort
```

Comment:
Actually, it looks like the only reason `Array(*args)` works here and elsewhere in this file is that `DependsOn#load` always calls the writer method with a single argument (hence the name `arg` rather than `args`), which in this case is an array.

So if a cask had `depends_on macos: ['10.8', '10.9']`, this method would get called with `send(:macos=, ['10.8', '10.9'])`, so `args` would be `[['10.8', '10.9']]`, and `Array(*args)` would be `['10.8', '10.9']`.

The best way to disambiguate this, I think, would be to remove the splat from `arg` in the method signature and explicitly convert it to an array at the beginning of the method with `Array(*arg)`. Same applies to the other `key=` methods.